nemo/collections/common/data/lhotse/cutset.py

```diff
@@ -330,7 +330,7 @@ def parse_and_combine_datasets(
         if (w := item.get("weight")) is not None:
             weights.append(w)
 
-    assert all(t == tarred_status[0] for t in tarred_status), "Mixing tarred and non-tarred datasets is not supported."
+    # assert all(t == tarred_status[0] for t in tarred_status), "Mixing tarred and non-tarred datasets is not supported."
```

Comment:
This is commented out to allow mixing tarred and untarred datasets, then use `force_map_dataset` or `force_iterable_dataset` later to control whether they're wrapped into map or iterable dataset. @pzelasko Do we have a better solution now?